enterprise/frontend/src/metabase-enterprise/sandboxes/components/MappingEditor.tsx

```diff
@@ -74,34 +118,37 @@ export const MappingEditor = ({
         </thead>
       ) : null}
       <tbody>
-        {entries.map(([key, value], index) => {
+        {entries.map(({ key, value }, index) => {
           const keyInput = renderKeyInput({
             value: key,
             placeholder: keyPlaceholder,
             onChange: newKey =>
-              onChange(replaceMappingKey(mapping, key, newKey)),
+              handleChange(replaceEntryKey(entries, index, newKey)),
+            error: entryError(entries, key),
           });
           const valueInput = renderValueInput({
             value: value,
             placeholder: valuePlaceholder,
             onChange: newValue =>
-              onChange(replaceMappingValue(mapping, key, newValue)),
+              handleChange(replaceEntryValue(entries, index, newValue)),
           });
           return (
             <tr key={index}>
-              <td className="pb1">
+              <td className="pb1" style={{ verticalAlign: "bottom" }}>
```

Comment:
Admittedly, inline styling isn't great. However, we don't have a class for verticalAlign, and I thought adding one would be worse. But if needed, this can be a styled component.

I did check and see if Mantine's table component would help, but it's pretty minimal in v6 (it gets better in v7), and doesn't include anything for vertical alignment